presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PredicatePushDown.java

```diff
@@ -490,6 +490,28 @@ private OuterJoinPushDownResult processLimitedOuterJoin(Expression inheritedPred
                     combineConjuncts(postJoinConjuncts.build()));
         }
 
+        private boolean shouldRemoveOuterJoinConjunct(Collection<Symbol> outerSymbols, Expression conjunct)
+        {
+            if (!(conjunct instanceof ComparisonExpression)) {
```

Comment:
if conjuct is not ComparisonExpression or at least it's not equality comparison and it touches both sides of the table I believe it should still stay? Or not? Can this change cause a deoptimization? Query like:

```
FILTER (a2 = b2)
  (some complicated query)
    LOJ(a1=b1 AND a2 <= b2)
      TS(t1(a1, a2))
      TS(t2(b1, b2)
```

Could be rewritten by this change to:

```
FILTER (a2 = b2)
  (some complicated operators)
    LOJ(a1=b2)
      TS(t1(a1, a2))
      TS(t2(b1, b2)) and FILTER b2 <= b2
```

which semantically seams ok, but might return more rows from LOJ (which will be filtered out later by FILTER) but still it can cause more work in "(some complicated operators)". Unless we will relay that this `FILTER (a2 = b2)` will be pushed down again in another pass... But if I'm not seriously mistaken (complicated code...) previous version with checking just for `usesOnlySymbols` should be safer.